### PR TITLE
Fix devcontainer url (closing #35145)

### LINKED ISSUE #35145: [Bug]: Codespaces URL is wrong [DEVOPS]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

After opening the `nextcloud/server` project inside GitHub Codepsaces and setting up the new NC instance, the following error occures after trying to access the Web UI:

```
Fastly error: unknown domain: <REMOVED>2a-80.apps.codespaces.githubusercontent.com. Please check that this domain has been added to a service.

Details: cache-<REMOVED>
```

image

This is due to a wrong redirection inside `codespace.config.php` after successfully installed the new instance.

### Steps to reproduce

1. Open new Codespace from GitHub UI
2. Navigate to NC webapplication
3. Setup database credentials and install the instance
4. Try to access the NC webapplication again

### Expected behavior

NC should open correctly inside GitHub Codespaces.

### Installation method

_No response_

### Operating system

_No response_

### PHP engine version

_No response_

### Web server

_No response_

### Database engine version

_No response_

### Is this bug present after an update or on a fresh install?

_No response_

### Are you using the Nextcloud Server Encryption module?

_No response_

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

```shell
<NOT IMPORTANT>
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by R0Wi**
Fix #35145 

See also:
* https://github.com/community/community/discussions/5104#discussioncomment-2217052
* https://docs.github.com/en/codespaces/developing-in-codespaces/default-environment-variables-for-your-codespace#list-of-default-environment-variables

### CODE REVIEW

File: .devcontainer/codespace.config.php [hunk 1]
```diff
@@ -12,7 +13,7 @@
     'memcache.local' => '\OC\Memcache\APCu',
 ];
 
-if($cloudEnvironmentId !== true) {
-    $CONFIG['overwritehost'] = $cloudEnvironmentId . '-80.apps.codespaces.githubusercontent.com';
+if($codespaceName !== true) {
```
Comment (kesselb) [inline .devcontainer/codespace.config.php:16]: `getenv` will return a string or false. I find this check for not true a bit weird. I guess this should be "not empty"?

**Review by kesselb — COMMENTED** [review]
Thank you!

File: .devcontainer/codespace.config.php (hunk 1 above)
Comment (R0Wi) [reply .devcontainer/codespace.config.php]: @kesselb added appropriate checks for both variables which should be clearer, I think 👍

Comment (kesselb) [reply .devcontainer/codespace.config.php]: Please ;)

**Review by kesselb — APPROVED** [review]
Thank you

**Review by szaimen — APPROVED** [review]
🐘

**Comment by R0Wi** [comment]
Seems like unrelated drone tests are failing

```
Scenario: sharee can not reshare a file with edit permission if the sharer disables it # /drone/src/tests/acceptance/features/app-files-sharing.feature:402 | 797s
-- | --
491 | Given I act as John                                                                  # ActorContext::iActAs() | 797s
492 | And I am logged in as the admin                                                      # LoginPageContext::iAmLoggedInAsTheAdmin() | 805s
493 | And I act as Jane                                                                    # ActorContext::iActAs() | 805s
494 | And I am logged in                                                                   # LoginPageContext::iAmLoggedIn() | 810s
495 | And I act as John                                                                    # ActorContext::iActAs() | 810s
496 | And I rename "welcome.txt" to "farewell.txt"                                         # FileListContext::iRenameTo() | 811s
497 | And I see that the file list contains a file named "farewell.txt"                    # FileListContext::iSeeThatTheFileListContainsAFileNamed() | 811s
498 | And I share "farewell.txt" with "user0"                                              # FilesAppSharingContext::iShareWith() | 814s
499 | And I see that the file is shared with "user0"                                       # FilesAppSharingContext::iSeeThatTheFileIsSharedWith() | 814s
500 | And I set the share with "user0" as not editable                                     # FilesAppSharingContext::iSetTheShareWithAsNotEditable() | 1079s
501 | Share with user0 menu in the details view in Files app could not be found after 100 seconds | 1079s
502 | Allow editing checkbox input in the share with user0 menu in the details view in Files app could not be found after 100 seconds (NoSuchElementException) | 1079s
503 | And I see that "user0" can not edit the share                                        # FilesAppSharingContext::iSeeThatCanNotEditTheShare() | 1079s
504 | When I act as Jane                                                                   # ActorContext::iActAs() | 1079s
505 | And I open the Files app                                                             # FilesAppContext::iOpenTheFilesApp() | 1079s
506 | And I share "farewell.txt" with "user1"                                              # FilesAppSharingContext::iShareWith() | 1079s
507 | Then I see that the file is shared with "user1"                                      # FilesAppSharingContext::iSeeThatTheFileIsSharedWith() | 1079s
508 | And I see that "user1" can not edit the share                                        # FilesAppSharingContext::iSeeThatCanNotEditTheShare() | 1079s
509 | And I see that "user1" can not be allowed to edit the share                          # FilesAppSharingContext::iSeeThatCanNotBeAllowedToEditTheShare() | 1079s
510 |   | 1081s
511 | --- Failed scenarios: | 1081s
512 |   | 1081s
513 | /drone/src/tests/acceptance/features/app-files-sharing.feature:402
```

@kesselb is there a possibility to restart these test or just to bypass them for this particular PR?

**Comment by R0Wi** [comment]
@szaimen @kesselb does anyone know how to restart the failing tests at https://drone.nextcloud.com/nextcloud/server/25740/58/4 ? Seems like I do not have permissions to do so ...

**Pull request merged**